javascript-practice/src/js/services/commonService.js

```diff
@@ -7,13 +7,23 @@ export default class CommonService {
     this.firebaseService = FirebaseService;
   }
 
+  /**
+   * Connect to Firebase Databse
+   */
   connectToDb() {
     this.firebaseService.reconnect();
   }
 
-  async findKeyByProperty(property, value, path = this.defaultPath) {
+  /**
+   * Find the id of data on database
+   * @param {string} property The property want to get value
+   * @param {string} value The value of property
+   * @param {path} path The path of datbase
+   * @returns {string} The id of data object
+   */
+  async findIdByProperty(property, value, path = this.defaultPath) {
     this.connectToDb();
-    const existUser = this.firebaseService.findKeyByPropery(
+    const existUser = this.firebaseService.findIdByProperty(
```

Comment:
It worked on your case. However, IMO, I think it's not good at all. Some frameworks use findRecordByProp or findFirstByProp or findFirstOrDefaultByProp, it return record, not id

Reply:
Fixed.